build-settings-logic/src/main/kotlin/DokkaBuildSettingsProperties.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import org.gradle.api.initialization.Settings
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.findByType
+import javax.inject.Inject
+
+/**
+ * Shared properties in `settings.gradle.kts` files.
+ *
+ * Fetch an instance using [DokkaBuildSettingsProperties.Companion.dokkaBuildSettingsProperties].
+ */
+abstract class DokkaBuildSettingsProperties @Inject constructor(
+    private val providers: ProviderFactory
+) {
+    val buildingOnTeamCity: Boolean = System.getenv("TEAMCITY_VERSION") != null
+    val buildingOnCi: Boolean = System.getenv("CI") != null || buildingOnTeamCity
+
+
+    //region Gradle Build Scan
+    // NOTE: build scan properties are documented in CONTRIBUTING.md
+    /** If unset, the user has not opted in to publish Build Scans. */
+    val buildScanUrl: Provider<String> =
+        dokkaProperty("build.scan.url")
+
+    /** Optionally override the default name attached to a Build Scan. */
+    val buildScanUsername: Provider<String> =
+        dokkaProperty("build.scan.username")
+            .orElse(BUILD_SCAN_USERNAME_DEFAULT)
+            .map(String::trim)
+    //endregion
+
+
+    //region Gradle Build Cache
+    val buildCacheLocalEnabled: Provider<Boolean> =
+        dokkaProperty("build.cache.local.enabled", String::toBoolean)
+            .orElse(!buildingOnCi)
+            .orElse(true)
```

Comment:
not sure, that this line will be ever executed, as `orElse` above will set value, isn't it?
(the same for `buildCachePushEnabled`)